test/CurrencyReserves.t.sol

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.20;
+
+import {CurrencyReserves} from "../src/libraries/CurrencyReserves.sol";
+import {Test} from "forge-std/Test.sol";
+import {Currency} from "../src/types/Currency.sol";
+
+contract CurrencyReservesTest is Test {
+    using CurrencyReserves for Currency;
+
+    Currency currency0;
+
+    function setUp() public {
+        currency0 = Currency.wrap(address(0xbeef));
+    }
+
+    function test_getReserves_returns_set() public {
+        currency0.syncCurrencyAndReserves(100);
+        uint256 value = CurrencyReserves.getSyncedReserves();
+        assertEq(value, 100);
+        assertEq(Currency.unwrap(CurrencyReserves.getSyncedCurrency()), Currency.unwrap(currency0));
+    }
+
+    function test_set_twice_returns_correct_value() public {
+        currency0.syncCurrencyAndReserves(100);
+        currency0.syncCurrencyAndReserves(200);
+        uint256 value = CurrencyReserves.getSyncedReserves();
+        assertEq(value, 200);
+        assertEq(Currency.unwrap(CurrencyReserves.getSyncedCurrency()), Currency.unwrap(currency0));
+    }
+
+    function test_reset_currency() public {
+        currency0.syncCurrencyAndReserves(100);
+        uint256 value = CurrencyReserves.getSyncedReserves();
+        assertEq(value, 100);
+        assertEq(Currency.unwrap(CurrencyReserves.getSyncedCurrency()), Currency.unwrap(currency0));
+        CurrencyReserves.reset();
+        uint256 valueAfterReset = CurrencyReserves.getSyncedReserves();
+        assertEq(valueAfterReset, 100);
```

Comment:
yeah definitely dont like this - would love it to also be reset to 0 please

Reply:
Its strictly never used by the contract and 100 extra gas

Reply:
We should just be checking that that value is not used after another call to sync

Reply:
I also don't love the extra gas, but I also don't love we leave it unset. What if we read both in the transient state library and if currency == 0, we return 0 for reserves?

Reply:
Agree, that would make it clear for anyone using the TransientStateLibrary library.
And either the extra gas is used in the settle for deleting the currency, or alternative in the TransientStateLibrary read, but that one should probably have lower prio then the settle?
Also for the CurrendyReserves `getReserves` function, it was renamed to `getSyncedReserves`, so the name helps to indicates the sync condition I would say.

Reply:
Plus anyone using the TransientStateLibrary could decide themselves to remove it if they want to save the gas in the read :)

Reply:
I like this suggestion for TransientStateLibrary